### PR TITLE
Collapse linked sources if more than 5 entries are included

### PR DESCRIPTION
**Pull request by restylianos**
### Changes
- Collapse **linked sources** if they contain more than 5 entries
- Fix missing div keys after the use of Array.prototype.map
- Lint fixes (make lint introduced several changes in python files)
- Isolate link components and SimpleView
- Remove unused vars and console logs

### CODE REVIEW

**Comment by restylianos** [comment]
> Hey @restylianos , thanks a lot for the PR! Can you please fix the linting?

Hi, @northdpole it should be ok now. Please note that 
**[local setup]**
`black --version
black, 24.1a1 
Python (CPython) 3.10.12` 
**[super-linter@v4]**
`black==22.12.0`

File: application/frontend/src/components/DocumentNode/DocumentNode.tsx [hunk 1]
```diff
@@ -61,119 +79,55 @@ export const DocumentNode: FunctionComponent<DocumentNode> = ({
...
+                {idx > 0 && <hr style={{ borderColor: 'transparent', margin: '20px 0' }} />}
+                <div>
+                  <b>Which {getDocumentTypeText(type, links[0].document.doctype, node.doctype)}</b>:
+                  {/* Risk here of mixed doctype in here causing odd output */}
+                </div>
+                <div>
+                  <div className="accordion ui fluid styled f0">
+                    {sortedResults.map((link, i) => {
+                      const temp = (
+                        <div key={`document-node-container-${type}-${idx}-${i}`}>
```
Comment (northdpole) [inline application/frontend/src/components/DocumentNode/DocumentNode.tsx:111]: good idea making the key unique but referenceable

File: application/frontend/src/pages/Search/components/BodyText.tsx [hunk 2]
```diff
@@ -18,9 +18,11 @@ export const SearchBody = () => {
           Use the search bar or <a href="/root_cres">browse the catalogue of all top-level topics</a>, try
           <a href="/node/standard/OWASP%20Top%2010%202021"> the Top10 2021 page</a> and click around, or{' '}
           <a href="/search/session">search for "Session"</a>, or check out{' '}
-          <a href="/cre/764-507"> CRE 764-507 </a> or <a href="/cre/581-525">CRE 581-525</a> to see the <b>Multilink</b>
-          feature: one link to OpenCRE gives access to a wide range of standards and related topics: testing advice, 
-          development, in-depth technical information, threat descriptions, articles, and tool settings.
+          <a href="/cre/764-507"> CRE 764-507 </a> or <a href="/cre/581-525">CRE 581-525</a> to see the{' '}
```
Comment (northdpole) [inline application/frontend/src/pages/Search/components/BodyText.tsx:21]: just wondering: how's the {' '} different than the whitespace that was there before

File: application/tests/cre_main_test.py [hunk 3]
```diff
@@ -378,7 +378,9 @@ def test_parse_standards_from_spreadsheeet(self) -> None:
                 "Standard CWE (from ASVS)-hyperlink": "https://example.com/cwe19876",
                 "Link to other CRE": "FooBar",
                 "Standard NIST 800-53 v5": "SA-22 Unsupported System Components",
-                "Standard NIST 800-53 v5-hyperlink": "https://example.com/nist-800-53-v5",
```
Comment (northdpole) [inline application/tests/cre_main_test.py:381]: is this change necessary?
it's just adding parentheses


same for all changes in this file

File: application/tests/defs_test.py [hunk 4]
```diff
@@ -83,7 +83,9 @@ def test_document_todict(self) -> None:
                                 "document": {
                                     "doctype": "Standard",
                                     "name": "ASVS",
-                                    "section": "SESSION-MGT-TOKEN-DIRECTIVES-DISCRETE-HANDLING",
```
Comment (northdpole) [inline application/tests/defs_test.py:86]: is the parentheses addition necessary here?

File: application/tests/spreadsheet_parsers_test.py [hunk 5]
```diff
@@ -754,7 +754,9 @@ def test_parse_hierarchical_export_format(self) -> None:
                 "Standard CWE (from ASVS)-hyperlink": "https://example.com/cwe19876",
                 "Link to other CRE": "FooBar",
                 "Standard NIST 800-53 v5": "SA-22 Unsupported System Components",
-                "Standard NIST 800-53 v5-hyperlink": "https://example.com/nist-800-53-v5",
```
Comment (northdpole) [inline application/tests/spreadsheet_parsers_test.py:757]: are the changes in this file necessary?

File: application/tests/web_main_test.py [hunk 6]
```diff
@@ -77,15 +77,18 @@ def test_extend_cre_with_tag_links(self) -> None:
                 )
             ),
             "cc": cres["cc"],
-            "cd": cres["cd"]
-            .add_link(
-                defs.Link(
-                    ltype=defs.LinkTypes.PartOf, document=cres["ca"].shallow_copy()
+            "cd": (
```
Comment (northdpole) [inline application/tests/web_main_test.py:80]: are the changes in this file necessary or relevant to the issue being worked on?

**Review by northdpole — CHANGES_REQUESTED** [review]
@restylianos , thanks a ton for the work on this, you have some good ideas and your implementation seems clean.

Unfortunately it seems that some autoformater somewhere went rogue so there's a lot more changes than necessary or relevant to the issue being worked on. 

Do you think you could undo the irrelevant changes to the *.py files? Then happy to approve your PR

**Comment by northdpole** [comment]
> > Hey @restylianos , thanks a lot for the PR! Can you please fix the linting?
> 
> Hi, @northdpole it should be ok now. Please keep in mind that there is an inconsistency between the development environment and the lint code base workflow **[local setup]** `black --version black, 24.1a1 Python (CPython) 3.10.12` **[super-linter@v4]** `black==22.12.0`

huh, good catch!
Guess the super linter is a couple major versions behind, still it seems to be backwards compatible so this does not seem to be a problem for now

**Comment by restylianos** [comment]
@northdpole Sure I will drop the python changes asap and I will also investigate the suggested improvements related to eslint.

**Comment by restylianos** [comment]
@northdpole Hi, I just dropped the python changes. 

#### Workflow fail and unnecessary python file changes
The python changes were created after running make lint locally. The first workflow fail (4 files), was related to the inconstancy between my environment and super-linter@v4 which lead to a different result. Downgrading black to version 22.12.0 solved this issue and running make lint does no longer format additional python files. 

#### Eslint
> thanks for removing unused imports :-). if you know how we can do this automatically or add this to linting it'd be great! 

This could be achieved by introducing eslint to the lint process . The yarn run lint command **currently leads to the execution of prettier** ` "lint": "prettier --write '**/*.tsx' '**/*.ts' '**/*.js'",` . Including eslint to the lint process introduced too many changes related to 23 {\*.ts,\*.tsx} files  and dev dependencies. The following output was produced after a POC version `✖ 194 problems (111 errors, 83 warnings)`. I would be happy to work on the required changes, however, I think that this should be a different issue in order to avoid mixing issues up, in a single pr.

**Pull request closed without merging**